### PR TITLE
1866706 - Usage the Term Description to enrich the swagger document

### PR DESCRIPTION
**Pull request by cmclopes**
**1866706:**
Change on swagger generator to bring entity property term description an enrich the information present on the open API document

### CODE REVIEW

**Comment by cmclopes** [comment]
Hi @ssethupathi can you review this form me after the merge of #501

File: interaction-media-hal/src/main/java/com/temenos/interaction/media/hal/HALProvider.java [hunk 1]
```diff
@@ -360,11 +361,17 @@ else if (object instanceof OCollection) {
...
-						&& property.getValue() != null) {
-						map.put(simpleName, buildFromOObject(entityMetadata,
-															 qualifiedName,
-															 property.getValue()));
+					if (entityMetadata.getPropertyVocabulary(qualifiedName) != null) {
+					    
+					    boolean mandatory = false;
+		                
+		                if(null!=entityMetadata.getPropertyVocabulary(qualifiedName).getTerm(TermMandatory.TERM_NAME)) {
+		                    mandatory = ((TermMandatory) entityMetadata.getPropertyVocabulary(qualifiedName).getTerm(TermMandatory.TERM_NAME)).isMandatory();
```
Comment (ssethupathi) [inline interaction-media-hal/src/main/java/com/temenos/interaction/media/hal/HALProvider.java]: Can we not use?
entityMetadata.getTermValue(property, termName)

File: interaction-media-hal/src/main/java/com/temenos/interaction/media/hal/HALProvider.java [hunk 2]
```diff
@@ -387,9 +394,16 @@ protected void buildFromOEntity(Map<String, Object> map, OEntity entity, String
...
 
 			String simpleName = simpleOPropertyName(entityMetadata, property);
-			if (entityMetadata.getPropertyVocabulary(simpleName) != null
-				&& property.getValue() != null) {
-				map.put(simpleName, buildFromOObject(entityMetadata, simpleName, property.getValue()));
+			if (entityMetadata.getPropertyVocabulary(simpleName) != null) {
+			    boolean mandatory = false;
+			    
+			    if(null!=entityMetadata.getPropertyVocabulary(simpleName).getTerm(TermMandatory.TERM_NAME)) {
+			        mandatory = ((TermMandatory) entityMetadata.getPropertyVocabulary(simpleName).getTerm(TermMandatory.TERM_NAME)).isMandatory();
```
Comment (ssethupathi) [inline interaction-media-hal/src/main/java/com/temenos/interaction/media/hal/HALProvider.java]: Same as above

**Comment by cmclopes** [comment]
Hi @ssethupathi , change done. Can you review and merge please?

**Pull request merged**